apps/docs/components/content/code.tsx

```diff
@@ -31,209 +30,148 @@ export function FocusLines({ children, lines }: { children: ReactNode; lines: nu
 	return <FocusLinesContext.Provider value={lines}>{children}</FocusLinesContext.Provider>
 }
 
-const blurredLineClassName = 'opacity-50 !text-white'
+const blurredLineClassName = '[&_*]:!opacity-60 [&_*]:!text-white'
 
 function CodeLink({ children, href }: { children: ReactNode; href: string }) {
 	return (
-		<A href={href} className="group-[.not-prose]:hover:underline">
-			<LinkIcon className="h-3.5 mb-1 mr-0.5 inline-block" />
+		<A
+			href={href}
+			className="group-[.not-prose]:underline group-[.not-prose]:hover:no-underline decoration-[#667e8b]"
+		>
 			{children}
 		</A>
 	)
 }
 
-export function Code({ children, ...props }: React.ComponentProps<'code'>) {
-	const codeLinks = useContext(CodeLinksContext)
-	const focusLines = useContext(FocusLinesContext)
+// does this token look like a JS identifier?
+function isIdentifier(token: string): boolean {
+	return /^[_$a-zA-Z\xA0-\uFFFF][_$a-zA-Z0-9\xA0-\uFFFF]*$/g.test(token)
+}
 
-	const newChildren = useMemo(() => {
-		// to linkify code, we have to do quite a lot of work. we need to take the output of
-		// Highlight.js and transform it to add hyperlinks to certain tokens. There are a few things
-		// that make this difficult:
-		//
-		// 1, the structure is recursive. A function span will include a bunch of other spans making
-		// up the whole definition of the function, for example.
-		//
-		// 2, a given span doesn't necessarily correspond to a single identifier. For example, this
-		// code: `dispatch: (info: TLEventInfo) => this` will be split like this:
-		// - `dispatch`
-		// - `: (info: TLEventInfo) => `
-		// - `this`
-		//
-		// That means we need to take highlight.js's tokens and split them into our own tokens that
-		// contain single identifiers to linkify.
-		//
-		// 3, a single identifier can be split across multiple spans. For example,
-		// `Omit<Geometry2dOptions>` will be split like this:
-		// - `Omit`
-		// - `<`
-		// - `Geometry2`
-		// - `dOptions`
-		// - `>`
-		//
-		// I don't know why this happens, it feels like a bug. We handle this by keeping track of &
-		// merging consecutive tokens if they're identifiers with no non-identifier tokens in
-		// between.
-
-		// does this token look like a JS identifier?
-		function isIdentifier(token: string): boolean {
-			return /^[_$a-zA-Z\xA0-\uFFFF][_$a-zA-Z0-9\xA0-\uFFFF]*$/g.test(token)
+// split the code into an array of identifiers, and the bits in between them
+function tokenize(code: string): string[] {
+	const identifierRegex = /[_$a-zA-Z\xA0-\uFFFF][_$a-zA-Z0-9\xA0-\uFFFF]*/g
```

Comment:
i see this predates this PR but can you have the `isIdentifier` fn above use this same regex? so it's not duplicated

Reply:
they're not the same though, `isIdentifier` includes the `^` and `$` sentinels to match a whole string, that wouldn't work here.

Reply:
ah! i didn't notice that, my bad. well, they could be composed:

`new RegExp('^' + identifierRegexWithoutSentinels.source + '$')`